### PR TITLE
[JUJU-2431] Install dependencies for github runner

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following updates the static-analysis job for the static-analysis.

## QA steps

Tests pass

### CODE REVIEW

File: tests/suites/static_analysis/lint_go.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 run_go() {
 	VER=$(golangci-lint --version | tr -s ' ' | cut -d ' ' -f 4 | cut -d '.' -f 1,2)
-	if [[ ${VER} != "1.46" ]] && [[ ${VER} != "v1.46" ]]; then
+	if [[ ${VER} != "1.50" ]] && [[ ${VER} != "v1.50" ]]; then
 		(echo >&2 -e '\nError: golangci-lint version does not match 1.46. Please upgrade/downgrade to the right version.')
```
Comment (jack-w-shaw) [inline tests/suites/static_analysis/lint_go.sh:4]: Need to update error msg

File: cmd/jujud/agent/machine_test.go [hunk 2]
```diff
@@ -131,14 +125,7 @@ func (s *MachineSuite) SetUpTest(c *gc.C) {
...
 func bootstrapRaft(c *gc.C, dataDir string) {
-	err := raftworker.Bootstrap(raftworker.Config{
-		Clock:      clock.WallClock,
-		StorageDir: filepath.Join(dataDir, "raft"),
-		LocalID:    "0",
-		Logger:     loggo.GetLogger("machine_test.raft"),
-		Queue:      queue.NewOpQueue(clock.WallClock),
-	})
-	c.Assert(err, jc.ErrorIsNil)
+	// TODO (stickupkid): Use the dqlite suite.
```
Comment (SimonRichardson) [inline cmd/jujud/agent/machine_test.go:128]: @manadart I know you're working on this.

File: cmd/jujud/agent/machine_test.go [hunk 3]
```diff
@@ -1412,52 +1399,7 @@ func (s *MachineSuite) TestModelWorkersRespectSingularResponsibilityFlag(c *gc.C
...
-		transport,
-	)
-	c.Assert(err, jc.ErrorIsNil)
-	defer sink.Close()
-	err = newSnapshot.Persist(sink)
-	if err != nil {
-		sink.Cancel()
-	}
-	c.Assert(err, jc.ErrorIsNil)
+	// TODO(stickupkid): Implement for the dqlite implementation.
```
Comment (SimonRichardson) [inline cmd/jujud/agent/machine_test.go:1402]: @manadart I know you're working on this.

File: worker/lease/store.go [hunk 4]
```diff
@@ -92,7 +92,7 @@ AND    l.name = ?`
 	}
 
 	result, err := leasesFromRows(rows)
-	return result, errors.Trace(rows.Err())
+	return result, err
```
Comment (SimonRichardson) [inline worker/lease/store.go:95]: @manadart This is why we need linters :roll_eyes:

File: worker/leaseexpiry/worker_test.go [hunk 5]
```diff
@@ -70,6 +70,7 @@ INSERT INTO lease (uuid, lease_type_id, model_uuid, name, holder, start, expiry)
 VALUES (?, 1, 'some-model-uuid', ?, ?, datetime('now'), datetime('now', ?))`[1:]
 
 	stmt, err := s.DB.Prepare(q)
+	c.Assert(err, jc.ErrorIsNil)
```
Comment (SimonRichardson) [inline worker/leaseexpiry/worker_test.go:73]: Another one we missed

File: .github/golangci-lint.config.yaml [hunk 6]
```diff
@@ -32,9 +32,9 @@ linters:
     - ineffassign
     - misspell
     - unconvert
-    - deadcode
-    - varcheck
```
Comment (SimonRichardson) [inline .github/golangci-lint.config.yaml:36]: These are deprecated, they've been surpassed with `unused`, but `unused` is a beast and one we should tackle in the future, but not now.

**Review by manadart — APPROVED** [review]

**Pull request merged**